One side note. On a recent version of python (3.11) `make init-dev`fails to build  (`make init` works fine)

```
pyamplipi/amplipi.py:536: error: Incompatible default for argument "timeout" (default has type "None", argument has type "int")  [assignment]
```

The fix is to update the announcement definition in `pyamplipi/amplipi.py` to 
``` async def announce(self, announcement: Announcement, timeout: Optional[int] = None) -> Status:```

--- a/.github/workflow/sphinx-docs.yml
+++ b/.github/workflow/sphinx-docs.yml
@@ -32,11 +32,7 @@ jobs:
 
       - name: Build documentation
         run: |
-          cd docs
-          # Generate API documentation
-          sphinx-apidoc -o . ../pyamplipi -f
-          # Build HTML docs
-          make html
+          make docs
           touch _build/html/.nojekyll
 
       - name: Deploy to GitHub Pages

--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@ PYTHON = python3
 TEST_PATH = ./tests/
 FLAKE8_EXCLUDE = venv,.venv,.eggs,.tox,.git,__pycache__,*.pyc
 
-.PHONY: clean init init-de test check build
+.PHONY: clean init init-dev test check build docs
 
 clean:
 	@find . -name '*.pyc' -exec rm --force {} +
@@ -14,6 +14,7 @@ clean:
 	@rm -rf *.egg-info
 	@rm -f *.sqlite
 	@rm -rf .cache
+	@rm -rf docs/_build
 
 init:
 	@pip install --upgrade pip
@@ -32,3 +33,7 @@ check:
 
 build:
 	@${PYTHON} setup.py build
+
+docs:
+	@cd docs && sphinx-apidoc -o . ../pyamplipi -f
+	@$(MAKE) -C docs html

--- a/docs/index.rst
+++ b/docs/index.rst
@@ -14,5 +14,4 @@ Indices and tables
 
 * :ref:`genindex`
 * :ref:`modindex`
-* :ref:`search`
  

--- a/pyamplipi/amplipi.py
+++ b/pyamplipi/amplipi.py
@@ -533,7 +533,7 @@ class AmpliPi:
         return Status.parse_obj(response)
 
     # -- anounce call
-    async def announce(self, announcement: Announcement, timeout: int = None) -> Status:
+    async def announce(self, announcement: Announcement, timeout: Optional[int] = None) -> Status:
         """
         Announce a message using the AmpliPi system.
 

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -1,3 +1,6 @@
 pytest>=7.2.0
 flake8>=6.0.0
 mypy>=0.991
+sphinx>=8.1.3
+sphinx-rtd-theme>=3.0.1
+myst-parser>=4.0.0

